application/apps/indexer/session/src/state/indexes/keys.rs

```diff
@@ -0,0 +1,91 @@
+use crate::events::{NativeError, NativeErrorKind};
+use indexer_base::progress::Severity;
+
+#[derive(Debug)]
+pub struct Keys {
+    keys: Vec<u64>,
+    sorted: bool,
+}
+
+impl Keys {
+    pub fn new() -> Self {
+        Keys {
+            keys: vec![],
+            sorted: false,
+        }
+    }
+    pub fn add(&mut self, position: u64) {
+        self.keys.push(position);
+        self.sorted = false;
+    }
+
+    pub fn remove(&mut self, position: &u64) {
+        if let Ok(index) = self.keys.binary_search(position) {
+            self.keys.remove(index);
+        }
+    }
```

Comment:
not correct:
``` rust
mod test {
    #[test]
    fn test_keys() {
        let mut keys = super::Keys::new();
        assert_eq!(keys.first(), None);
        keys.add(13);
        keys.add(100);
        keys.add(12);
        keys.add(11);
        keys.add(200);
        keys.add(1);
        keys.remove(&100);
        let res = keys.clone();
        assert_eq!(res, vec![1, 11, 12, 13, 200]);
    }
}
```
will fail

Reply:
yes, sure. But this test doesn't reflect real use-case and context. Our streams are increasing only. Well as for new keys we always will get it incrementally from 1 to 2. Exception - bookmarks (in these keys we do sorting for sure). But in anyway I've added extra-checks for sorted or not.